src/components/composite/MarginAmount/MarginAmount.tsx

```diff
@@ -38,7 +39,23 @@ const MarginAmount: React.FunctionComponent<MarginAmountProps> = ({
       return defaultVal.toString();
     }
   };
+  
+  const swapForm = useSwapFormContext();
+  const positionForm = usePositionContext();
 
+  let marginAmount;
+  if (underlyingTokenName?.includes('USD')) {
+    const portfolioMarginInfo = positionForm.position?.margin.toString();
+    marginAmount = !isUndefined(portfolioMarginInfo) ? parseInt(portfolioMarginInfo)/1e6 : defaultMargin;
+  } else {
+    const portfolioMarginInfo = positionForm.position?.margin.toString();
+    marginAmount = !isUndefined(portfolioMarginInfo) ? parseInt(portfolioMarginInfo) / 1e18 : defaultMargin;
+  }
```

Comment:
```suggestion
  const divisor = underlyingTokenName?.includes('USD') ? 1e6 : 1e18;
  const portfolioMarginInfo = positionForm.position?.margin.toString();
  const marginAmount = !isUndefined(portfolioMarginInfo) ? parseInt(portfolioMarginInfo)/divisor : defaultMargin;
```

Reduce duplication of code 😁

Reply:
I think that works well and the withdrawable margin amounts seems reasonable. Happy to merge if you give green light. @ssbarbee

Reply:
Yes, we checked this just now in our peer review and the information seems to match up. Worth making a ticket for unifying the sources of margin information as currently PortfolioContext and PositionContext serve very similar information.